src/pages/Main/Main.tsx

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react';
+import CardList from '../../components/CardList/CardList';
+import CardSearch from '../../components/CardSearch/CardSearch';
+import Loader from '../../components/Loader/Loader';
+import NotFoundData from '../../components/NotFound/NotFoundData';
+import Pagination from '../../components/Pagination/Pagination';
+import useFetching from '../../hooks/useFetching';
+import { useQueryParams } from '../../hooks/useQueryParams';
+import styles from './Main.module.css';
+import ErrorBtn from '../../components/Error/ErrorBtn/ErrorBtn';
+
+export default function Main() {
+  const [value, setValue] = useState('');
+  const {
+    queryPage,
+    queryLimit,
+    querySearch,
+    searchValue,
+    setDefaultQueryParametr,
+    setSearchParams,
+  } = useQueryParams();
+  const [limitPageItem, setLimitPageItem] = useState(Number(queryLimit));
+  const [lastClickTime, setLastClickTime] = useState(0);
+  const { loading, data, pageInfo, fetchAllCards, setPageInfo } = useFetching();
+
+  const getCards = async (value: string, page: number, limit: number) => {
+    fetchAllCards(value, page, limit);
+
+    setSearchParams({
+      page: String(page || pageInfo.currPage) || queryPage,
+      q: value || querySearch || searchValue,
+      limit: String(limit || setDefaultQueryParametr(queryLimit, '10')),
+    });
+  };
+
+  useEffect(() => {
+    if (searchValue) {
+      setValue(searchValue);
+    }
+    getCards(
+      querySearch,
+      setDefaultQueryParametr(queryPage, '1'),
+      setDefaultQueryParametr(queryLimit, '10')
+    );
+  }, []);
+
+  const handleInputValueChange = (value: number) => {
+    getCards(querySearch, 1, value);
+    setLimitPageItem(value);  
+  };
+
+  const onPageChange = useCallback(
+    (newPage: number) => {
+      const now = Date.now();
+      if (now - lastClickTime >= 600) {
```

Comment:
What will happen if this line would be removed?

Reply:
looks not really clear to me

Reply:
I`ll change it